### PR TITLE
add t2-reboot cli command

### PR DESCRIPTION
**Pull request by dbuentello**
completes RFC `t2 reboot`

### CODE REVIEW

**Comment by dbuentello** [comment]
Took a stab at writing tests, feedback appreciated.

File: lib/tessel/commands.js [hunk 1]
```diff
@@ -179,3 +179,8 @@ module.exports.reconnectDnsmasq = function() {
 module.exports.reconnectDhcp = function() {
   return ['/etc/init.d/odhcpd', 'restart'];
 };
+
+module.exports.reboot = function() {
+  return ['sh', '-c', 'echo 39 > /sys/class/gpio/export && echo out > /sys/class/gpio/gpio39/direction && echo 0 > /sys/class/gpio/gpio39/value && reboot'];
```
Comment (rwaldron) [inline lib/tessel/commands.js]: Can just copy these: https://github.com/tessel/t2-firmware/blob/master/node/tessel-export.js#L143-L154

File: lib/controller.js [hunk 2]
```diff
@@ -1026,6 +1026,17 @@ controller.tesselEnvVersions = opts => {
   });
 };
 
+controller.reboot = opts => {
+  opts.authorized = true;
+  return controller.standardTesselCommand(opts, tessel => {
+    return tessel.reboot().then(() => {
+      log.info('Tessel Rebooted');
```
Comment (dbuentello) [inline lib/controller.js]: Things get racy here...
- I'd say _still rebooting_ (will reword output)
- Yes `reboot` and power cycle are different.

A power cycle is SoC and Coprocessor clearing their states. `reboot` only resets the SoC's state. See [reset button](https://github.com/tessel/t2-firmware/issues/185#issuecomment-244577542) issue.

File: lib/tessel/commands.js (hunk 1 above)
Comment (dbuentello) [inline lib/tessel/commands.js]: Added comments

File: lib/tessel/reboot.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+// Internal
+var commands = require('./commands');
+var Tessel = require('./tessel');
+
+/*
+ Reboots Tessel
+ */
+Tessel.prototype.reboot = function() {
+  return new Promise((resolve,reject) => {
+    return this.simpleExec(commands.reboot()).then(resolve).catch(reject);
```
Comment (dbuentello) [inline lib/tessel/reboot.js]: Fixed. Promises are new to me so I appreciate comments like this. Also, discovered a bug! :)

**Comment by dbuentello** [comment]
Having issues with gracefully terminating cli and Promises https://github.com/tessel/t2-cli/pull/985/commits/7510c951ae8fe95a6bd2876defd0f8288a84a06d

**Comment by dbuentello** [comment]
Looks good! 
```sh
dbuentello@Daniels-MacBook-Pro:~/_repos/Github/dbuentello/t2-cli (feature/t2-reboot)$ node bin/tessel-2.js reboot --usb
INFO Looking for your Tessel...
INFO Connected to Tessel-02A3F368A7CF.
INFO Tessel Rebooting...
dbuentello@Daniels-MacBook-Pro:~/_repos/Github/dbuentello/t2-cli (feature/t2-reboot)$ node bin/tessel-2.js reboot --lan
INFO Looking for your Tessel...
INFO Connected to Tessel-02A3F368A7CF.
INFO Tessel Rebooting...
```

**Pull request merged**